src/main/java/edu/cornell/kfs/sys/service/BatchFileDirectoryService.java

```diff
@@ -0,0 +1,15 @@
+package edu.cornell.kfs.sys.service;
+
+import org.kuali.kfs.sys.businessobject.SystemOptions;
+import org.kuali.rice.core.api.util.KeyValue;
+import org.springframework.cache.annotation.Cacheable;
+
+import java.io.File;
+import java.util.List;
+
+public interface BatchFileDirectoryService {
+
+    @Cacheable(value= SystemOptions.CACHE_NAME, key="'buildDirectoryKeyValuesList' + #p0")
```

Comment:
That matches the parameter to this method (0 for first; in this case there is only one). So the cache key includes the parameter passed to the method so different parameters will have different cache entries.